backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/ArtifactStoreService.java

```diff
@@ -98,7 +99,11 @@ default String getUrl(
     UriComponentsBuilder uriComponentsBuilder =
         ServletUriComponentsBuilder.newInstance().scheme(scheme).host(finalHost).path(endpoint);
     for (Map.Entry<String, Object> queryParam : parameters.entrySet()) {
-      uriComponentsBuilder.queryParam(queryParam.getKey(), queryParam.getValue());
+      if (queryParam.getKey().equals(ModelDBConstants.FILENAME)) {
+        uriComponentsBuilder.path(String.valueOf(queryParam.getValue()));
+      } else {
+        uriComponentsBuilder.queryParam(queryParam.getKey(), queryParam.getValue());
+      }
```

Comment:
A comment here will be helpful, something like 
"Adding the filename to the path to enable file saved with that name."